src/prefect/agent/local/agent.py

```diff
@@ -147,21 +146,7 @@ def deploy_flow(self, flow_run: GraphQLResult) -> str:
             )
             raise TypeError("Unsupported Storage type: %s" % type(storage).__name__)
 
-        # If the flow is using a run_config, load it
-        if getattr(flow_run.flow, "run_config", None) is not None:
-            run_config = RunConfigSchema().load(flow_run.flow.run_config)
-            if not isinstance(run_config, LocalRun):
-                self.logger.error(
-                    "Flow run %s has a `run_config` of type `%s`, only `LocalRun` is supported",
-                    flow_run.id,
-                    type(run_config).__name__,
-                )
-                raise TypeError(
-                    "Unsupported RunConfig type: %s" % type(run_config).__name__
-                )
-        else:
-            run_config = None
-
+        run_config = self._get_run_config(flow_run, LocalRun)
```

Comment:
Is there a reason why this one doesn't also have an `assert` like the other agents?

Reply:
we disable `mypy` on this file for some reason. The `assert` is just to appease mypy.